pkgs/by-name/di/digikam/package.nix

```diff
@@ -1,149 +1,213 @@
 
-  src = fetchurl {
-    url = "mirror://kde/stable/${pname}/${version}/digiKam-${version}-1.tar.xz";
-    hash = "sha256-BbFF/38vIAX6IbxXnBUqsjyBkbZ4/ylEyPBAbWud5tg=";
+  testData = fetchgit {
+    url = "https://invent.kde.org/graphics/digikam-test-data.git";
+    rev = "d02dd20b23cc279792325a0f03d21688547a7a59";
+    fetchLFS = true;
+    hash = "sha256-SvsmcniDRorwu9x9OLtHD9ftgquyoE5Kl8qDgqi1XdQ=";
+  };
+in
+
+mkDerivation (finalAttrs: {
+  pname = "digikam";
+  version = "8.4.0";
+
+  src = fetchFromGitLab {
+    domain = "invent.kde.org";
+    owner = "graphics";
+    repo = "digikam";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-GJYlxJkvFEXppVk0yC9ojszylfAGt3eBMAjNUu60XDY=";
   };
 
-  strictDeps = true;
+  patches = [ ./disable-tests-download.patch ];
 
-  depsBuildBuild = [ cmake ];
+  strictDeps = true;
 
   nativeBuildInputs = [
     cmake
-    doxygen
+    ninja
     extra-cmake-modules
-    libsForQt5.kdoctools
-    libsForQt5.wrapQtAppsHook
-    wrapGAppsHook3
-  ] ++ lib.optionals cudaSupport (with cudaPackages; [
-    cuda_nvcc
-  ]);
-
-  buildInputs = [
-    bison
-    boost
-    eigen
-    exiv2
-    ffmpeg_4
     flex
-    graphviz
-    imagemagick
-    lcms2
-    lensfun
-    libgphoto2
-    libheif
-    liblqr1
-    libusb1
-    libGL
-    libGLU
-    opencv
-    pcre
-    x265
-    jasper
-  ] ++ (with libsForQt5; [
-    libkipi
-    libksane
-    libqtav
-
-    qtbase
-    qtxmlpatterns
-    qtsvg
-    qtwebengine
-    qtnetworkauth
-
-    akonadi-contacts
-    kcalendarcore
-    kconfigwidgets
-    kcoreaddons
-    kfilemetadata
-    knotifications
-    knotifyconfig
-    ktextwidgets
-    kwidgetsaddons
-    kxmlgui
-
-    breeze-icons
-    marble
-    oxygen
-    threadweaver
-  ]) ++ lib.optionals cudaSupport (with cudaPackages; [
-    cuda_cudart
-  ]);
+    bison
+    qt6Packages.wrapQtAppsHook
+    wrapGAppsHook3
+  ] ++ lib.optionals cudaSupport [ cudaPackages.cuda_nvcc ];
+
+  # Based on <https://www.digikam.org/api/index.html#externaldeps>,
+  # but it doesn’t have everything, so you also have to check the
+  # CMake files…
+  #
+  # We list non‐Qt dependencies first to override Qt’s propagated
+  # build inputs.
+
+  buildInputs =
+    [
+      opencv
+      libtiff
+      libpng
+      # TODO: Figure out how on earth to get it to pick up libjpeg8 for
+      # lossy DNG support.
+      libjpeg
+      libheif
+      libjxl
+      boost
+      lcms2
+      expat
+      exiv2
+      libxml2
+      libxslt
+      # Qt WebEngine uses and propagates FFmpeg, and if it’s a
+      # different version it causes linker warnings.
+      #ffmpeg
+      jasper
+      eigen
+      lensfun
+      liblqr1
+      libgphoto2
+      libusb1
+      imagemagick
+      x265
+
+      qt6Packages.qtbase
+      qt6Packages.qtnetworkauth
+      qt6Packages.qtscxml
+      qt6Packages.qtsvg
+      qt6Packages.qtwebengine
+      qt6Packages.qt5compat
+      qt6Packages.qtmultimedia
```

Comment:
we can get all these from `kdePackages`

Reply:
Is there a specific reason to, rather than using the canonical name?

Reply:
I don't see the benefit of passing in `kdePackages` and `qt6Packages`, making overrides more difficult?

Reply:
Fair enough. Overrides are already a bit awkward in the `pkgs/by-name` world where you don’t have scopes, but that’s a good enough reason for me. I’ll change it.

Reply:
Done.